dbt/adapters/snowflake/connections.py

```diff
@@ -443,25 +443,28 @@ def _split_queries(cls, sql):
         split_query = snowflake.connector.util_text.split_statements(sql_buf)
         return [part[0] for part in split_query]
 
-    @classmethod
-    def process_results(cls, column_names, rows):
-        # Override for Snowflake. The datetime objects returned by
-        # snowflake-connector-python are not pickleable, so we need
-        # to replace them with sane timezones
-        fixed = []
+    @staticmethod
+    def _fix_rows(rows: Iterable[Iterable]) -> Iterable[Iterable]:
```

Comment:
@mikealfare This approach makes sense to me, but I would also like to get this change out tomorrow. Would you be open to dealing with the new override structure in a follow-up?

Regarding the return type, every Iterator is also Iterable, so I believe it is correct. And it is also as non-specific as possible, which I would argue is good, since it gives us the widest degree of flexibility in what we accept and return.

Reply:
This is what I mean:

```
>>> issubclass(Iterator, Iterable)
True
```

Reply:
> Would you be open to dealing with the new override structure in a follow-up?

Yes, this is fine. We should probably discuss the generic approach before doing it anyway. It's just a low hanging beam that we constantly bonk our heads on, so I always tend towards fixing it.

> Regarding the return type, every Iterator is also Iterable, so I believe it is correct. And it is also as non-specific as possible, which I would argue is good, since it gives us the widest degree of flexibility in what we accept and return.

Sure, both are correct. I didn't intend to say that `Iterable` is incorrect. I was simply arguing that `Iterator` better reflects what's being done in this method and the story in general. I agree that argument types should be more general to allow for more flexible implementations, as long as they capture the intended requirements of the argument. Along those same lines, the return type should capture the intended usage of the result. I understood that intended usage to be something that was lazy loaded to improve memory usage, especially given the usage of `yield` and the fact that this method will never be overridden by a downstream application. If that's not the case, then `Iterable` is fine.